### DISCUSSION TITLE
Nanopore Sequencing Foundation Model

### DISCUSSION THREAD

**Discussion by CCranney** [xomicsdatascience/AttentionSmithy#13] [Bioinformatics]
For context on this field of nanopore sequencing generally, [here's a paper in Nature](https://www.nature.com/articles/s41587-021-01108-x) that describes what it is and why it is important. I'll be providing my own summary, though I'll use their Figure 1.

Nucleotide sequencing (DNA sequencing) has gone through several phases of development over the years, but a more recent and exciting area is nanopore sequencing. By all comparisons it is a cheap, portable, fast, and straightforward way to get DNA sequence data. The main problem is its inaccuracy - there's the raw data straight from the machine, but converting that into a DNA sequence with high accuracy has proven challenging. And this is an area we absolutely need precision for - if 1/100 of your sequenced nucleotides are wrong, and you sequence billions of nucleotides... you get the idea. You need a margin of error that is infinitesimally small if you're going to rely on it. We've come a long way, but it's still a struggle. We have a lot of nanopore raw data, but not a lot of verified nanopore-to-sequence data to train on from the deep learning perspective.

The raw data is basically a 1D datastream. Nucleotide sequences are run through a pore, there's an electrical current running nearby, and that current is jiggled differently depending on the nucleotides that are passing through. We are looking at bumps in a line and using that to make a nucleotide sequence, like what is shown in Figure 1 of the above-referenced paper:

![Screenshot 2025-02-13 at 12 04 54 PM](https://github.com/user-attachments/assets/47fefb3f-e6de-475e-b15f-8e2062c3f2f3)

From that perspective, it is very similar to speech recognition tasks. In those instances, you have a 1D datastream and use that to interpret words (or, more technically, linguistic characters). There have been proposals of various kinds to train a foundation model on raw speech audio, then fine-tune it for rarely-used languages (see [this review here](http://arxiv.org/abs/2205.10643)). This way you can use speech from any language (in pre-training), then refine the output to be specific to your desired use case (in fine-tuning).

Why not do the same for nanopore sequencing? What if we made a nanopore sequencing foundation model that was pre-trained on all the nanopore sequencing data we could find (no labels necessary!!), then fine-tuned for specific genome sequencing tasks? 

If that works and we made nanopore sequencing more viable for broader applications, it would be difficult to fully grasp the implications. I advise you to look up a nanopore sequencing device, it is literally the size of a thumb drive. It would make reliable sequencing cheap, convenient, portable, and accessible.
